File: nodejs/tests/adc_dac/hx711_test.js
'use strict';

const Gpio = require('onoff').Gpio;
const { HX711Transport } = require('../../packages/periph/src/transport/hx711');
const { HX711Full }      = require('../../packages/periph/src/chips/adc_dac/hx711');

const DOUT_PIN   = parseInt(process.env.HX711_DOUT   || '5',  10);
const PD_SCK_PIN = parseInt(process.env.HX711_PD_SCK || '6',  10);

let passed = 0;
let failed = 0;

function checkTrue(label, condition) {
    if (condition) { console.log('PASS', label); passed++; }
    else           { console.log('FAIL', label); failed++; }
}

const dout   = new Gpio(DOUT_PIN,   'in');
const pd_sck = new Gpio(PD_SCK_PIN, 'out');
const transport = new HX711Transport(dout, pd_sck);
const chip = new HX711Full(transport);

checkTrue('isReady returns boolean', typeof chip.isReady() === 'boolean');

const raw = chip.readRaw();
checkTrue('readRaw returns number', typeof raw === 'number');
checkTrue('readRaw in 24-bit signed range', raw >= -8388608 && raw <= 8388607);

chip.setGain(128);
checkTrue('setGain(128) accepted', true);

chip.setGain(64);
checkTrue('setGain(64) accepted', true);

chip.setGain(32);
checkTrue('setGain(32) accepted', true);

chip.setGain(128);

try {
    chip.setGain(99);
    checkTrue('setGain(99) throws', false);
} catch (e) {
    checkTrue('setGain(99) throws', true);
}

const avg = chip.readAverage(3);
checkTrue('readAverage returns number', typeof avg === 'number');
checkTrue('readAverage in 24-bit signed range', avg >= -8388608 && avg <= 8388607);

chip.tare(3);
checkTrue('tare accepted', true);

const offset = chip.getOffset();
checkTrue('getOffset returns number', typeof offset === 'number');

chip.setScale(420.0);
checkTrue('setScale accepted', true);

const scale = chip.getScale();
checkTrue('getScale returns 420.0', scale === 420.0);

const weight = chip.readWeight(1);
checkTrue('readWeight returns number', typeof weight === 'number');

chip.powerDown();
checkTrue('powerDown accepted', true);

chip.powerUp();
checkTrue('powerUp accepted', true);

transport.close();
checkTrue('close accepted', true);

console.log(`===DONE: ${passed} passed, ${failed} failed===`);
process.exit(failed === 0 ? 0 : 1);

File: python/tests/adc_dac/hx711_test.py
import _testconfig as cfg
from machine import Pin
from periph.transport.hx711_micropython import HX711Transport
from periph.chips.adc_dac.hx711 import HX711Full

passed = 0
failed = 0


def check_true(label, condition):
    global passed, failed
    if condition:
        print('PASS', label)
        passed += 1
    else:
        print('FAIL', label)
        failed += 1


dout   = Pin(cfg.DOUT,   Pin.IN)
pd_sck = Pin(cfg.PD_SCK, Pin.OUT)
transport = HX711Transport(dout, pd_sck)
chip = HX711Full(transport)

check_true('is_ready returns bool', isinstance(chip.is_ready(), bool))

raw = chip.read_raw()
check_true('read_raw returns int', isinstance(raw, int))
check_true('read_raw in 24-bit signed range', -8388608 <= raw <= 8388607)

chip.set_gain(128)
check_true('set_gain(128) accepted', True)

chip.set_gain(64)
check_true('set_gain(64) accepted', True)

chip.set_gain(32)
check_true('set_gain(32) accepted', True)

chip.set_gain(128)

try:
    chip.set_gain(99)
    check_true('set_gain(99) raises ValueError', False)
except ValueError:
    check_true('set_gain(99) raises ValueError', True)

avg = chip.read_average(3)
check_true('read_average returns int', isinstance(avg, int))
check_true('read_average in 24-bit signed range', -8388608 <= avg <= 8388607)

chip.tare(3)
check_true('tare accepted', True)

offset = chip.get_offset()
check_true('get_offset returns int', isinstance(offset, int))

chip.set_scale(420.0)
check_true('set_scale accepted', True)

scale = chip.get_scale()
check_true('get_scale returns float', isinstance(scale, float))
check_true('get_scale returns 420.0', scale == 420.0)

weight = chip.read_weight(1)
check_true('read_weight returns float', isinstance(weight, float))

chip.power_down()
check_true('power_down accepted', True)

chip.power_up()
check_true('power_up accepted', True)

print('===DONE: {} passed, {} failed==='.format(passed, failed))
